misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt

```diff
@@ -134,6 +137,7 @@ class DockerVitessCluster(
   private var containerId: String? = null
 
   private var isRunning = false
+  private var stopContainerOnExit = true
```

Comment:
which test will actually stop the container? If I'm reading this correctly, the first test will start up the container and also have `stopContainerOnExit`, meaning when it ends it'll shut down the container as well?

Reply:
This will stop the container on JVM exit, as the `stop()` method is called by the JVM shutdown hook. `stopContainerOnExit` is set to `false` if the docker container was not started by this JVM, to allow it to keep running across test runs.

Reply:
Actually let me add a little detail for posterity:

All Vitess clusters used by the app/test are tracked by `StartVitessService` in a global cache (look for the `CacheBuilder`). The cache is invalidated by a JVM shutdown hook. On invalidation, the cache's `removalListener` will call the each entry's `DockerVitessCluster`'s `stop()` method. If `stopContainerOnExit` is set to true, it will stop and remove the container.

Reply:
Thanks! Want to add these details in the javadoc :-)

Reply:
Sure.